### ISSUE TITLE
search feature

### ISSUE THREAD

**Issue by pgonzale60** [lencx/ChatGPT#364]
### Feature description

Dear Chen ,

I would like to request the addition of a search feature within the application. This feature would allow users to search for specific keywords, phrases, or messages within their conversations, making it easier and quicker to locate the information they need.

Some key considerations for this feature could include (1) the ability to search across all conversations or within specific ones and (2) the ability to search for messages based on date ranges.

### Motivation

Currently, users can only browse through their conversations to find specific messages, but this can become time-consuming and inefficient when trying to locate specific information in a large number of conversations.

### Alternatives

Browse through the conversations.

**Comment by kn9** [comment]
I completely agree with this suggestion. A search feature would significantly improve the user experience and efficiency when searching for important information within their conversations. The additional considerations you mentioned, such as the ability to search across all conversations or within specific ones and based on date ranges, would make this feature even more versatile and useful. I believe this feature would be greatly appreciated by the users and could set this application apart from others.

**Comment by alexlyee** [comment]
bumping this! :)

**Comment by lencx** [comment]
The search needs to be supported by data, and at the moment it is just a wrapper for the official website, with no access to data.
